spec/features/rsvping_workshop.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+
+RSpec.describe 'a Coach', type: :feature do
+  describe '#workshop' do
+    let(:coach) { Fabricate(:coach) }
+    let(:future_workshop) { Fabricate(:workshop) }
+    let(:past_workshop) { Fabricate(:past_workshop) }
+
+    before do
+      login(coach)
+    end
+
+    it 'can rsvp to a workshop' do
+        visit workshop_path(future_workshop)
+        click_on 'Attend as a coach'
+        expect(page).to have_button('Attend')
+    end
+
+    it 'can not rsvp to a past workshop' do
+        visit workshop_path(past_workshop)
+        expect(page).to have_content('This event has already taken place')
+        expect(page).not_to have_button('Attend as a coach')
+    end
+
+    it 'can not rsvp to workshop that is now in the past' do
+      visit workshop_path(future_workshop)
+
+      travel_into_the_future = Time.zone.now + 3.days
+      allow(Time).to receive(:now).and_return(travel_into_the_future)
+
+      click_on 'Attend as a coach'
+      expect(page).to have_current_path(workshop_path(future_workshop), ignore_query: true)
+      expect(page).to have_content('This event has already taken place')
+      expect(page).not_to have_button('Attend as a coach')
+    end
+  end
+end
```

Comment:
We already have a spec testing this functionality https://github.com/codebar/planner/blob/master/spec/features/managing_workshop_attendance_spec.rb so I we only need the last scenario. It should probably be added to the [shared examples](https://github.com/codebar/planner/blob/master/spec/support/shared_examples/behaves_like_managing_workshop_attendance.rb) file.